src/__internal__/click-away-wrapper/click-away-wrapper.component.tsx

```diff
@@ -14,11 +19,13 @@ const ClickAwayWrapper = ({
   eventTypeId = "click",
   targets,
 }: ClickAwayWrapperProps) => {
+  const targetsRef = useRef(targets);
+
```

Comment:
```suggestion
targetsRef.current = targets
```

Reply:
The `targetsRef` needs to be updated on each rerender to make this pattern work.

Reply:
It isn't needed for any of the components that currently implement it, the only scenario I found that required it if the `targets` array were dynamically updated (adding/ removing/ changing the ref that points to an element). I'll add it in as we'll likely need it when/ if we add ref forwarding to them or if any new components do dynamically alter the `targets`.

https://codesandbox.io/s/beautiful-mayer-kkflin?file=/src/index.js